chain/x/oracle/common_test.go

```diff
@@ -69,3 +68,45 @@ func getTestOracleScript() (os types.OracleScript, clear func()) {
 		"sourceCodeURL",
 	), func() { deleteFile(filepath.Join(dir, filename)) }
 }
+
+// wat2wasm compiles the given Wat content to Wasm, relying on the host's wat2wasm program.
+func wat2wasm(wat []byte) []byte {
+	inputFile, err := ioutil.TempFile("", "input")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(inputFile.Name())
+	outputFile, err := ioutil.TempFile("", "output")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(outputFile.Name())
+	if _, err := inputFile.Write(wat); err != nil {
+		panic(err)
+	}
+	if err := exec.Command("wat2wasm", inputFile.Name(), "-o", outputFile.Name()).Run(); err != nil {
+		panic(err)
+	}
+	output, err := ioutil.ReadFile(outputFile.Name())
+	if err != nil {
+		panic(err)
+	}
+	return output
+}
+
+func mustGetOwasmCode(filename string) []byte {
+	absPath, _ := filepath.Abs(fmt.Sprintf("../../pkg/owasm/res/%s", filename))
```

Comment:
Let's move out of this legacy `pkg/owasm`. We can create `testfiles` folder in `x/oracle`